src/object/obj_tx.c

```diff
@@ -1701,6 +1719,17 @@ dc_tx_cpd_body_need_bulk(size_t size)
 	return dc_obj_proto_version > 8 && size >= DAOS_BULK_LIMIT;
 }
 
+static inline size_t
+dc_tx_cpd_adjust_size(size_t size)
+{
+	/* Lower layer (mercury) need some additional space to pack related payload into
+	 * RPC body (or specified buffer) via related proc interfaces, usually that will
+	 * not exceed 1/10 of the payload. We can make some relative large estimation if
+	 * we do not exactly know the real size now.
+	 */
+	return size * 11 / 10;
```

Comment:
"Lower layer (mercury) need some additional space to pack related payload "
I don't know about it , seems mercury will not insert extra data (except checksum which is disabled in your code).

maybe it is due to for example dc_tx_cpd_sub_reqs_size() is not very accurate, for example "size += sizeof(*dcu->dcu_bulks) * dcsr->dcsr_nr" the packed bulk handle size is more than that.

Reply:
Yes, that is what I mean "additional".